src/coreclr/vm/prestub.cpp

```diff
@@ -2297,7 +2297,17 @@ PCODE MethodDesc::DoPrestub(MethodTable *pDispatchingMT, CallerGCMode callerGCMo
             pCode = GetStubForInteropMethod(this);
         }
 
-        GetOrCreatePrecode();
+#ifdef FEATURE_PORTABLE_ENTRYPOINTS
+        // Store the IL Stub interpreter data on the actual
+        // P/Invoke MethodDesc.
+        void* ilStubInterpData = PortableEntryPoint::GetInterpreterData(pCode);
+        _ASSERTE(ilStubInterpData != NULL);
+        SetInterpreterCode((InterpByteCodeStart*)ilStubInterpData);
+#else // !FEATURE_PORTABLE_ENTRYPOINTS
+        // Only vararg P/Invoke use shared stubs, they need a precode to push the hidden argument.
+        if (IsVarArg())
+            GetOrCreatePrecode();
```

Comment:
```suggestion
```
It may make more sense to add `pNMD->GetOrCreatePrecode()` call at   https://github.com/dotnet/runtime/blob/f2b00689e5ab854bdb133014fd13f937fe1a1f76/src/coreclr/vm/dllimport.cpp#L5653-L5656. It will save IsVarArg check that is not cheap, and keep the varargs special handling in fewer places.

(I expect that we are not going to support varargs with `FEATURE_PORTABLE_ENTRYPOINTS`.)

Reply:
Good idea.